bindings/c/foundationdb/fdb_c.h

```diff
@@ -481,10 +481,27 @@ DLLEXPORT WARN_UNUSED_RESULT FDBFuture* fdb_transaction_get_mapped_range(FDBTran
                                                                          int target_bytes,
                                                                          FDBStreamingMode mode,
                                                                          int iteration,
-                                                                         int matchIndex,
                                                                          fdb_bool_t snapshot,
                                                                          fdb_bool_t reverse);
 
+DLLEXPORT WARN_UNUSED_RESULT FDBFuture* fdb_transaction_get_mapped_range_optional_index(FDBTransaction* tr,
```

Comment:
I was doing some digging to try and figure out how this is _supposed_ to work, and I think I figured it out:

1. To change a function signature, the old method should be renamed to `fdb_transaction_get_mapped_range_v719`, indicating that the old method should be used for API versions below 720.
2. The new method is that named `fdb_transaction_get_mapped_range_impl`. The old method should probably be changed to just call the new method, but with the new parameter set to a default value.
3. Add `FDB_API_CHANGED(fdb_transaction_get_mapped_range, 720)` to `fdb_select_api_version_impl` like so: https://github.com/apple/foundationdb/blob/e6b22547261f0e618c868336f0a111dbe22e44b0/bindings/c/fdb_c.cpp#L1137

The above steps should make it so that the old function is exposed when loaded from an older client. What I'm a little fuzzy on is whether newer clients (like, say, the FDB Java bindings) need to worry about both versions existing or not. Adding a new method like here is probably more straightforward, I suppose.

Reply:
will discuss offline, this is the PR that I abort https://github.com/apple/foundationdb/pull/9016/files